Respond actor on batch operation requests

korthout opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

Adds `actorType` and `actorId` to the batch operation endpoint responses.

Note that these fields are only provided for batch operations created since 8.9.

Note that this PR does not include an IT yet, as this requires:
- https://github.com/camunda/camunda/issues/42070

Filtering and sorting are excluded as this is covered by other issues:
- https://github.com/camunda/camunda/issues/42068
- https://github.com/camunda/camunda/issues/42069

## Related issues

closes #42067

korthout committed:
feat: add actor info to batch-operations spec

Allows all batch operation responses to provide actor type and actor id.

Note that the fields are only set for batch operations created on 8.9 or
later.

This is copied from AuditLogResponse, as we want to keep them separated
for now. If we feel it's useful to combine these into a single Actor -
define a single ActorTypeEnum for reuse then we can change that later
still.

korthout committed:
feat: respond actor info for batch operations

Any request that responds a batch operation (GET/Search) now responds
the actor type and actor id. Note that both can be null.

The type enum value is mapped like any other enum in this class.

korthout committed:
test: respond actor info for batch operations

These tests were failing because the actor info is now responded.

tmetzke commented on zeebe/gateway-protocol/src/main/proto/v2/batch-operations.yaml:
```diff
@@ -609,6 +616,12 @@ components:
         - UPDATE_VARIABLE
 
+    BatchOperationActorTypeEnum:
+      type: string
```
🔧 Let's add a description here for the enum. We might even move the description you added above to the attribute and simply use a simple ref at the attribute.

korthout replied:
@tmetzke I've copied this from `audit-logs.yaml` where the types don't have descriptions. If that's something we want to do, could we perhaps enforce it?

I'm happy to move the description (or copy it) for batch operations. Whatever you think is best. Let me know what you'd like me to do.

tmetzke replied:
I see, thanks, Nico 👍
We are missing too many descriptions in the spec right now to enforce it via linting. We first need to ramp up our spec files, unfortunately. They are currently just warnings that are not surfaced well in CI.

The best way to do this is to have descriptions for the types/schemas. No matter if objects, enums, or whatever. Then, when referencing them, ideally, you just use a `$ref` and that's it, the description is picked up from that object/enum/whatever. If you need to override it because your attribute/parameter needs to express something else, you can use the `allOf` + `description` at the attribute.

tmetzke commented on zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/mapper/search/SearchQueryResponseMapper.java:
```diff
@@ -709,6 +710,12 @@ public static BatchOperationResponse toBatchOperation(final BatchOperationEntity
+            ofNullable(entity.actorType())
+                .map(Enum::name)
+                .map(BatchOperationActorTypeEnum::fromValue)
+                .orElse(null))
```
❓ For the other enums above (`batchOperationType` and `state`), we just use `.<attribute>(<REST-enum>.fromValue(entity.<enumAttribute>().name()))`. Should we do the same here? If there is an inherent benefit in doing it the way you did now, should we adjust how we handle the other enum attributes?

tmetzke reviewed (approved):
👍 Looks good to me.
🔧 Added some minor suggestions and questions, nothing blocking, though.

korthout committed:
docs: move description to enum type

This seems to be common practise.

korthout committed:
fix: use $ref rather than allOf

Currently $ref does not support a description, but that will soon be
fixed in OpenAPI 3.1.x. To deal with that, we currently use allOf as a
workaround as this does allow setting description. However, that's only
needed when we actually set a description. As this field doesn't specify
a custom description (it relies on the type having a description), we
have no need to use allOf.

Note that switching to allOf later would not have been a breaking
change. So it's also not necessary to set this already in an attempt to
be flexible in the future in case OpenAPI doesn't fix this problem.

fabiopaini-camunda reviewed (approved):
👍 LGTM!
